app/components/connectionForm.js

```diff
@@ -13,7 +13,11 @@ export default class Connect extends Component {
     return (
       <div>
         <input ref="ip" id="ip" name="ip" placeholder="Ip Eg: 12.0.0.1/localhost"/>
-        <button type="button" className="btn btn-success" onClick={this.handleConnect}>Connect</button>
+        <button type="button" className="btn btn-success" onClick={this.handleConnect}>
+          {
+            this.props.status === 'CONNECTING' ? 'Connecting...' : 'Connect'
```

Comment:
You mean component level state?

On Wed, Mar 23, 2016 at 4:11 PM, Yuva <notifications@github.com> wrote:

> In app/components/connectionForm.js
> <https://github.com/code-mancers/rbkit-electron-redux/pull/9#discussion_r57138716>
> :
>
> > @@ -13,7 +13,11 @@ export default class Connect extends Component {
> >      return (
> >        <div>
> >          <input ref="ip" id="ip" name="ip" placeholder="Ip Eg: 12.0.0.1/localhost"/>
> > -        <button type="button" className="btn btn-success" onClick={this.handleConnect}>Connect</button>
> > +        <button type="button" className="btn btn-success" onClick={this.handleConnect}>
> > +          {
> > +            this.props.status === 'CONNECTING' ? 'Connecting...' : 'Connect'
>
> won't a state change suffice?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/code-mancers/rbkit-electron-redux/pull/9/files/b5077fdb7adf104551977f5c9565030567154b5e#r57138716>
>